octodns/provider/mythicbeasts.py

```diff
@@ -0,0 +1,420 @@
+#
+#
+#
+
+from __future__ import absolute_import, division, print_function, \
+    unicode_literals
+
+import re
+
+from requests import Session
+from logging import getLogger
+
+from ..record import Record
+from .base import BaseProvider
+
+
+def add_trailing_dot(value):
+    '''
+    Add trailing dots to values
+    '''
+    assert value, 'Missing value'
+    assert value[-1] != '.', 'Value already has trailing dot'
+    return value + '.'
+
+
+def remove_trailing_dot(value):
+    '''
+    Remove trailing dots from values
+    '''
+    assert value, 'Missing value'
+    assert value[-1] == '.', 'Value already missing trailing dot'
+    return value[:-1]
+
+
+class MythicBeastsProvider(BaseProvider):
+    '''
+    Mythic Beasts DNS API Provider
+
+    mythicbeasts:
+      class: octodns.provider.mythicbeasts.MythicBeastsProvider
+        zones:
+          my-zone: 'password'
```

Comment:
The code below makes it look like `passwords` is required and has to be a `dict`, but this has `zones`.